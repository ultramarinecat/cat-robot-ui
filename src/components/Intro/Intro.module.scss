@charset 'utf-8';

////
//// Intro styles
////

@import 'base.scss';

.description {
  font-size: rem(28.8);
  line-height: 1.25;
  margin-bottom: rem(26);
  @media #{$tablet-plus} {
    font-size: rem(32);
  }
}
