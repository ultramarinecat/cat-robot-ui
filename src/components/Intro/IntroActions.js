//
// Intro actions
//

import { push } from 'connected-react-router';

import { AUTH } from '../Router/Router.js';

export const INTRO_CONTINUE = 'INTRO_CONTINUE';

export function introContinueAsync() {
  return dispatch => {
    dispatch(introContinue());

    dispatch(push(AUTH));
  };
}

function introContinue() {
  return {
    type: INTRO_CONTINUE
  };
}
