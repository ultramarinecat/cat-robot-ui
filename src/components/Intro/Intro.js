//
// Intro
// App landing page
//

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { withTranslation } from 'react-i18next';

import { introContinueAsync } from './IntroActions.js';
import Button from '../Button/Button.js';

import styles from './Intro.module.scss';

export function Intro({ introContinue, t }) {
  return (
    <div>
      <h2 className={styles.description}>{t('app_description')}</h2>
      <Button text={t('get_started')} onClick={introContinue} />
    </div>
  );
}

Intro.propTypes = {
  introContinue: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired
};

export default connect(
  null,
  { introContinue: introContinueAsync }
)(memo(withTranslation()(Intro)));
