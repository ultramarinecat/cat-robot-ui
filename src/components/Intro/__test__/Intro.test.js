import React from 'react';
import { render, fireEvent } from 'react-testing-library';

import { Intro } from '../Intro.js';

const CONTINUE = 'get_started';

describe('Intro', () => {
  it('should display a continue button', () => {
    const { container, getByText } = render(
      <Intro introContinue={jest.fn()} t={key => key} />
    );

    expect(getByText(CONTINUE)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call `introContinue` prop when continue button is clicked', () => {
    const introContinue = jest.fn();
    const { getByText } = render(<Intro introContinue={introContinue} t={key => key} />);

    fireEvent.click(getByText(CONTINUE));
    expect(introContinue).toHaveBeenCalled();
  });
});
