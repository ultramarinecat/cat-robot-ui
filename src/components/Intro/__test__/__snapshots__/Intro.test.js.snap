// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro should display a continue button 1`] = `
<div>
  <h2
    class="description"
  >
    app_description
  </h2>
  <button
    class="main waves-effect waves-light"
    type="button"
  >
    get_started
  </button>
</div>
`;
